lib/completion.dart

```diff
@@ -63,6 +67,13 @@ class DartCompleter extends CodeCompleter {
         String displayString = completion.isMethod ? '${completion.text}()' : completion.text;
         if (completion.returnType != null) displayString += ': ${completion.returnType}';
 
+        // Filter unmatching completions.
+        if (delta > 0) {
+          if (!completion.text.startsWith(prefix)) {
+            return null;
```

Comment:
We're filtering here for cc suggestions that are wildly inaccurate. This does make the UX better. ; however, if we do this, we need to file a higher priority issue to track down the root cause. The cc engine should not be making these suggestions. I don't know if the issue is in our use of the analysis server or in the analysis server itself.